handlers/handlers.go

```diff
@@ -119,13 +119,24 @@ func filterFlexOverview(w http.ResponseWriter, req *http.Request, rc RenderClien
 			return
 		}
 
-		dim.Options = append(dims.Items[i].Options, options...)
-		dims.Items[i] = dim
+		dim.Options = append(filterDims.Items[i].Options, options...)
+		filterDims.Items[i] = dim
+	}
+
+	datasetDims, err := dc.GetVersionDimensions(ctx, accessToken, "", collectionID, filterJob.Dataset.DatasetID, filterJob.Dataset.Edition, fmt.Sprint(filterJob.Dataset.Version))
```

Comment:
We're making a lot of api calls in this handler do you think it's worth making some of these concurrent to improve performance?

Reply:
It's probably worth grouping the ones that not dependent on the other, I've got no idea how to implement concurrent api calls in Go though

Reply:
Discussed - will refactor at a later date